kernel/kernel-api/src/main/java/io/delta/kernel/internal/TableConfig.java

```diff
@@ -256,19 +269,19 @@ public static Map<String, String> validateProperties(
       Engine engine, Map<String, String> configurations) {
     Map<String, String> validatedConfigurations = new HashMap<>();
     for (Map.Entry<String, String> kv : configurations.entrySet()) {
-      String key = kv.getKey().toLowerCase(Locale.ROOT);
-      String value = kv.getValue();
-      if (key.startsWith("delta.")) {
-        TableConfig<?> tableConfig = VALID_PROPERTIES.get(key);
-        if (tableConfig != null) {
-          tableConfig.validate(engine, value);
-          validatedConfigurations.put(tableConfig.getKey(), value);
+        String key = kv.getKey().toLowerCase(Locale.ROOT);
+        String value = kv.getValue();
+        if (key.startsWith("delta.") && VALID_PROPERTIES.containsKey(key)) {
+          TableConfig<?> tableConfig = VALID_PROPERTIES.get(key);
+          if (tableConfig.editable) {
+              tableConfig.validate(engine, value);
+              validatedConfigurations.put(tableConfig.getKey(), value);
+          } else {
+            throw DeltaErrors.cannotModifyTableProperty(key);
+          }
         } else {
           throw DeltaErrors.unknownConfigurationException(key);
         }
-      } else {
-        throw DeltaErrors.unknownConfigurationException(key);
```

Comment:
@vkorukanti - does this mean users cannot set arbitrary table properties? that doesn't seem correct ... is that what delta spark does?

Reply:
Yeah, this needs to be changed. Initially, we didn't see any need for arbitrary properties, but recently, StreamNative has a use case where they want to store arbitrary properties and fetch them as well. We will address this shortly.

Reply:
Does this mean that we don't need VALID_PROPERTIES?
We just need to check if key starts with "delta."?

Reply:
We still need to check if it starts with `delta.` and it is . Because we may encounter properties related to protocol (start with `delta.`) that are added in future versions of the Delta protocol spec and not yet supported in Kernel.